crates/pipeline_manager/src/db/test.rs

```diff
@@ -1035,6 +1036,66 @@ pub(crate) fn limited_option_connector() -> impl Strategy<Value = Option<Connect
     })
 }
 
+/// Generate different resource configurations
+pub(crate) fn runtime_config() -> impl Strategy<Value = RuntimeConfig> {
```

Comment:
any reason why we don't just implement the Arbitrary trait for RuntimeConfig using this strategy?

e..g here is an example on how to implement the trait
```
impl<K: Arbitrary + 'static, R: Arbitrary + 'static> Arbitrary for ColumnLayer<K, R> {
    type Parameters = usize;

    /// An arbitrary implementation that upholds the ColumnLayer constraints.
    fn arbitrary_with(limit: Self::Parameters) -> Self::Strategy {
        (0..max(1, limit))
            .prop_flat_map(|size| {
                let keys = proptest::collection::vec(any::<K>(), size);
                let diffs = proptest::collection::vec(any::<R>(), size);
                let lower_bound = 0..=size;
                (keys, diffs, lower_bound)
            })
            .prop_map(|(keys, diffs, lower_bound)| ColumnLayer {
                keys,
                diffs,
                lower_bound,
            })
            .boxed()
    }
    type Strategy = BoxedStrategy<Self>;
}
```

Reply:
@gz the type is defined in a different crate (pipeline-types) and in the past, the tests in this crate weren't picking up the trait implementation from the other trait. You can see the comment I left there from last time.